File: Parser/Parsers/IdiotParser.cs
using System;
using Domain.Models;
using Domain.Models.Enums;

namespace Parser.Parsers;

internal class IdiotParser
{
    public static Schedule Parse()
    {
        var nurses = new List<Nurse>();
        var shifts = new Shift[21];

        var nurse1 = new Nurse("Jens", Grade.ONE);
        var nurGse2 = new Nurse("Emma", Grade.TWO);
        var nurse3 = new Nurse("Paolo", Grade.THREE);

        nurses.Add(nurse1);
        nurses.Add(nurse2);
        nurses.Add(nurse3);

        var requirements = new Dictionary<Grade, int>();
        requirements[Grade.ONE] = 1;
        requirements[Grade.TWO] = 2;
        requirements[Grade.THREE] = 4;

        for(int i = 0; i < shifts.Count(); i++)
        {
            if(i % 3 == 0)
            {
                shifts[i] = new Shift(requirements, true);
            }
            else
            {
                shifts[i] = new Shift(requirements, false);
            }
        }

        return new Schedule(shifts, nurses);
    }
}

File: Parser/Parsers/IdiotParser.py
from Domain.Models.Enums.Grade import *
from Domain.Models.Nurse import *
from Domain.Models.Shift import *
from Domain.Models.Schedule import *


class IdiotParser:
    def parse(self):
        nurses = []
        shifts = []

        nurse1 = Nurse(0, Grade.ONE)
        nurse2 = Nurse(1, Grade.TWO)
        nurse3 = Nurse(2, Grade.THREE)

        nurses.append(nurse1)
        nurses.append(nurse2)
        nurses.append(nurse3)

        requirements = {
            Grade.ONE: 1,
            Grade.TWO: 2,
            Grade.THREE: 3
        }
        i = 0
        while i < 21:
            if i % 3 == 0:
                shifts.append(Shift(requirements, True))
            else:
                shifts.append(Shift(requirements, False))

            i += 1

        return Schedule(shifts, nurses)
